notifications.bs

```diff
@@ -905,19 +964,11 @@ attribute's getter must return the result of the following steps:
     <li><p>Set <var>action</var>'s {{NotificationAction/icon}} to
     <var>entry</var>'s <a>action icon URL</a>.
 
-    <!-- XXX IDL dictionaries are usually returned by value, so don't need to be
-         immutable. But FrozenArray reifies the dictionaries to mutable JS
-         objects accessed by reference, so we explicitly freeze them.
-         It would be better to do this with IDL primitives instead of JS - see
-         https://www.w3.org/Bugs/Public/show_bug.cgi?id=29004 -->
```

Comment:
Good question about synchronously reflecting the selected URL... that might require an async IPC message in Chrome. I'll have to think about that one.

The general solution to avoid having to freeze dictionaries seems to be to instead return interfaces with readonly attributes. NotificationAction now is an interface that only has readonly attributes, and assigning a new value to `icon` has no effect, intentionally. The property value does not change, and internally nothing changes. No exception is thrown.